workspaces/ballerina/ballerina-visualizer/src/views/BI/HelperPaneNew/Views/Variables.tsx

```diff
@@ -0,0 +1,346 @@
+import {  useEffect, useMemo, useRef, useState } from "react"
+import { getPropertyFromFormField, HelperPaneVariableInfo, useFieldContext } from "@wso2/ballerina-side-panel"
+import FooterButtons from "../Components/FooterButtons"
+import DynamicModal from "../Components/Modal"
+import { FormGenerator } from "../../Forms/FormGenerator"
+import { ScrollableContainer } from "../Components/ScrollableContainer"
+import { FormSubmitOptions } from "../../FlowDiagram"
+import { URI } from "vscode-uri"
+import styled from "@emotion/styled"
+
+type VariablesPageProps = {
+    fileName: string;
+    debouncedRetrieveCompletions?: (value: string, property: ExpressionProperty, offset: number, triggerCharacter?: string) => Promise<void>;
+    onChange: (value: string, isRecordConfigureChange: boolean, shouldKeepHelper?: boolean) => void;
+    targetLineRange: LineRange;
+    anchorRef: React.RefObject<HTMLDivElement>;
+    handleOnFormSubmit?: (updatedNode?: FlowNode, openInDataMapper?: boolean, options?: FormSubmitOptions) => void;
+    selectedType?: CompletionItem;
+    filteredCompletions: CompletionItem[];
+    currentValue: string;
+    recordTypeField?: RecordTypeField;
+    isInModal?: boolean;
+    handleRetrieveCompletions: (value: string, property: ExpressionProperty, offset: number, triggerCharacter?: string) => Promise<void>;
+}
+
+const VariablesMoreIconContainer = styled.div`
+    display: flex;
+    min-wdith: 20px;
+    align-items: center;
+    justify-content: center;
+    padding: 4px;
+     &:hover {
+        background-color:  ${ThemeColors.ON_SURFACE_VARIANT};
+        cursor: pointer;
+    }
+`;
+
+type BreadCrumbStep = {
+    label: string;
+    replaceText: string
+}
+
+export const Variables = (props: VariablesPageProps) => {
+    const { fileName, targetLineRange, onChange, anchorRef, handleOnFormSubmit, selectedType, filteredCompletions, currentValue, recordTypeField, isInModal, handleRetrieveCompletions } = props;
+    const [searchValue, setSearchValue] = useState<string>("");
+    const { rpcClient } = useRpcContext();
+    const [isLoading, setIsLoading] = useState<boolean>(false);
+    const newNodeNameRef = useRef<string>("");
+    const [isModalOpen, setIsModalOpen] = useState<boolean>(false);
+    const [projectPathUri, setProjectPathUri] = useState<string>();
+    const [breadCrumbSteps, setBreadCrumbSteps] = useState<BreadCrumbStep[]>([{
+        label: "Variables",
+        replaceText: ""
+    }]);
+
+    const { field, triggerCharacters } = useFieldContext();
+
+    useEffect(() => {
+        getProjectInfo()
+    }, []);
+
+    useEffect(() => {
+        const triggerCharacter =
+            currentValue.length > 0
+                ? triggerCharacters.find((char) => currentValue[currentValue.length - 1] === char)
+                : undefined;
+
+        console.log("Trigger Character:", triggerCharacter);
+    }, []);
```

Comment:
do we need this triggerCharacter initialization here, seems in the next useEffect it is again set. L96